### PR TITLE
route mail via mailgun

### PR DESCRIPTION
**Pull request by irabinovitch**
### What does this PR do?

Once merged our postfix cookbook will support routing outbound mail through third parties, such as mailgun.org.  Whether postfix will send direct or relay through mailgun will currently depend on if the mailgun secrets have been deployed at /etc/postfix/sasl_passwd.db

We opted for mailgun as the provider in this case as our agreement with Rackspace includes 50K messages per month.   At this time I only plan to enable this on scale-www1 and scale-reg1, as they have fairly low volume.  

We should review if Mailman and PHPlist would benefit from this in the future and consider standardizing, once we have a sense of how many emails we send each month.

### Motivation

Many users have been reporting that mail from our CFP and other systems is not reaching them, or is ending up flagged as spam.  By routing through an established mail provider we believe we'll have better success in reaching our users.

### Testing Guidelines

- Upload scale-secrets/common/sasl_passwd.db to /etc/postfix/ on the taste-tester role you wish to validate with OR your local vagrant vm.
- Run either taste-taster or chefctl as appropriate.
- Send yourself an email via CLI tools such as mailx. (eg echo "testing all the things"| mail -r <from-email> -s "test subject" <to-email>
- Use the password reset function in drupal to have it send you an email.  
- Once both emails have been received validate the headers include DMARC, DKIM, and SPF passes; as well as X-Mailgun-* headers.

### CODE REVIEW

**Comment by irabinovitch** [comment]
CC @jaymzh @jhoblitt @rgolan

**Comment by irabinovitch** [comment]
@jaymzh FYI hoping to merge this Sunday. With only 48 hours left in the CFP we're getting lots of complaints about issues registering or resetting passwords.

File: cookbooks/scale_postfix/recipes/default.rb [hunk 1]
```diff
@@ -36,6 +38,13 @@
   notifies :restart, 'service[postfix]', :immediately
 end
 
+file '/etc/postfix/sasl_passwd.db' do
```
Comment (jaymzh) [inline cookbooks/scale_postfix/recipes/default.rb]: You're creating this file unconditionally... and then keying things off of whether it exists, it'll always exist given this rule, so your code doesn't make sense.

Comment (irabinovitch) [inline cookbooks/scale_postfix/recipes/default.rb]: removed the notify.

**Comment by jaymzh** [comment]
Not sure why you removed that notify - seems necessary to me?

**Pull request merged**